src/ess.jl

```diff
@@ -195,19 +195,21 @@ function mean_autocov(k::Int, cache::BDAESSCache)
 end
 
 """
-    ess_rhat(
-        [estimator,]
+    ess(
```

Comment:
This seems to be a case where type inference does not do well. The type of `ess(randn(100, 10, 4); type=:tail)` is inferred to be `Any` even though every method that can be called would all return `Vector{Float64}`. If we use `Base.@constprop :agressive` or `@inline` here, then `foo(x) = ess(x; type=:tail)` becomes type-inferrable, but still not the version with the keyword argument.

Reply:
I guess one alternative is to define the output objects in the outer methods and then the internal method are all mutating `_exp!`. But it feels like there should be a way to make this type-inferrable.